ql/ql/src/codeql_ql/StructuredLogs.qll

```diff
@@ -317,12 +321,137 @@ module KindPredicatesLog {
     Predicate getPredicate() { result = getPredicateFromPosition(this.getPosition()) }
   }
 
+  /** Gets the `index`'th event that's evaluated by `recursive`. */
+  private InLayer layerEventRank(ComputeRecursive recursive, int index) {
+    result =
+      rank[index + 1](InLayer cand, int startline, string filepath |
+        cand.getComputeRecursiveEvent() = recursive and
+        cand.hasLocationInfo(filepath, startline, _, _, _)
```

Comment:
Hmm... Is there no other way of recovering the ordering? Longer term, I was kinda hoping we could do away with location info for the logs (as the locations relation in the DB is _quite_ big once you include one of the structured logs).

Reply:
I'm not aware of any other way, unfortunately.